mlir/lib/Target/LLVMIR/ModuleImport.cpp

```diff
@@ -874,6 +874,24 @@ Attribute ModuleImport::getConstantAsAttr(llvm::Constant *constant) {
   return {};
 }
 
+FlatSymbolRefAttr
+ModuleImport::getOrCreateFakeSymbolName(llvm::GlobalVariable *globalVar) {
+  assert(globalVar->getName().empty() &&
+         "expected to work with a nameless global");
+  auto it = namelessGlobals.find(globalVar);
```

Comment:
ultra nit: can you use try_emplace here. That function returns an iterator and if the value has been inserted and hence it avoids the second hashmap lookup blow.